gitoxide-core/src/repository/commit.rs

```diff
@@ -39,6 +47,60 @@ pub fn verify(repo: gix::Repository, rev_spec: Option<&str>) -> Result<()> {
     Ok(())
 }
 
+pub fn sign(repo: gix::Repository, rev_spec: Option<&str>) -> Result<()> {
+    let rev_spec = rev_spec.unwrap_or("HEAD");
+    let commit = repo
+        .rev_parse_single(format!("{rev_spec}^{{commit}}").as_str())?
+        .object()?
+        .into_commit();
+
+    let mut cmd: std::process::Command = gix::command::prepare("gpg").into();
+    cmd.args([
+        "--keyid-format=long",
+        "--status-fd=2",
+        "--detach-sign",
+        "--sign",
+        "--armor",
+    ])
+    .stdin(Stdio::piped())
+    .stdout(Stdio::piped());
+    gix::trace::debug!("About to execute {cmd:?}");
+    let mut child = cmd.spawn()?;
+    child
+        .stdin
+        .take()
+        .expect("to be present")
+        .write_all(commit.data.as_ref())?;
+
+    if !child.wait()?.success() {
+        bail!("Command {cmd:?} failed");
+    }
+
+    let mut signed_data = Vec::new();
+    child
+        .stdout
+        .take()
+        .expect("to be present")
+        .read_to_end(&mut signed_data)?;
+
+    let object = repo
+        .rev_parse_single(format!("{rev_spec}^{{commit}}").as_str())?
+        .object()?;
+    let mut commit_ref = object.to_commit_ref();
+
+    let extra_header: Cow<'_, BStr> = Cow::Borrowed(signed_data.as_bstr());
+
+    // TODO:
+    // What do we want to do when there is already a signature?
+    commit_ref
+        .extra_headers
+        .push((BStr::new(SIGNATURE_FIELD_NAME), extra_header));
+
+    eprintln!("{commit_ref:?}");
```

Comment:
@Byron At this point, I would like to write the signed commit back to the filesystem. I search the codebase for hints on how this is usually done, but couldn’t really find anything. Would you be able to point me to an example?